dali/operators/util/get_property.cc

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/util/get_property.h"
+#include "dali/pipeline/data/types.h"
+
+namespace dali {
+
+DALI_SCHEMA(GetProperty)
+    .DocStr(
+        R"code(Returns a property of the tensor passed as an input.
+The type of the output will depend on the ``key`` of the requested property.)code")
+    .NumInput(1)
+    .NumOutput(1)
+    .AddArg("key",
+            R"code(Specifies, which property is requested. The following properties are supported:
+
+* ``"source_info"``:
+* ``"layout"``:
+)code",
+            DALI_STRING);
+
+
+DALI_REGISTER_OPERATOR(GetProperty, GetProperty<CPUBackend>, CPU);
+
+namespace detail {
+
```

Comment:
idea: I suggest you add a `StringLikeProperty` base class, that you could reuse for SourceInfo and Layout

Reply:
I though about it, however I believe there wouldn't be too big of a gain with such abstraction:
1. `GetShape` functions are different
2. `GetType` is the same
3. `FillOutput` is partially the same, however I believe it would need so much abstraction magic (because `GetLayout` doesn't return `std::string`, but a `TensorLayout`), that the duplicated code is more readable. On top of that, this duplication is our beloved operator bolier-plate, so it's really almost an idiom in out case ;)